qiskit_nature/second_q/circuit/library/initial_states/hartree_fock.py

```diff
@@ -254,10 +254,13 @@ def hartree_fock_bitstring_mapped(
     else:
         qubit_op = qubit_converter.map(bitstr_op)
 
+    if not isinstance(qubit_op, SparsePauliOp):
+        qubit_op = qubit_op.primitive
```

Comment:
This code I imagine could stay here even after PauliSumOp, as a possible return type from mapping, gets removed. We may even easliy forget to remove it as-is down the road. Would it be better though to invert the test

```
    if isinstance(qubit_op, PauliSumOp):
        qubit_op = qubit_op.primitive
```
since we are much more likely to search and remove PauliSumOp logic and indeed this code would fail on the import, like it would elsewhere, when PauliSumOp gets removed so we have to take care of this.

Reply:
Yeh that is a good point :+1: